components/03-modules/modal/modal.js

```diff
@@ -4,45 +4,82 @@ class Modal {
     this.init();
   }
 
-  toggleModal(modalElement) {
-    modalElement.classList.toggle('modal--active')
+  trap(e, modal) {
+    if (e.which == 27) {
+      this.closeModal(modal);
+    }
+    if (e.which == 9) {
+      let currentFocus = document.activeElement;
+      let totalOfFocusable = modal.focusableChildren.length;
+      let focusedIndex = modal.focusableChildren.indexOf(currentFocus);
+      if (e.shiftKey) {
+        if (focusedIndex === 0) {
+          e.preventDefault();
+          modal.focusableChildren[totalOfFocusable - 1].focus();
+        }
+      }
+      else {
+        if (focusedIndex == totalOfFocusable - 1) {
+          e.preventDefault();
+          modal.focusableChildren[0].focus();
+        }
+      }
+    }
+  }
+
+  openModal(modal) {
+    modal.focused = document.activeElement;
+    modal.el.classList.add('modal--active');
+    modal.focusableChildren = Array.from(modal.el.querySelectorAll(modal.focusable));
+    modal.focusableChildren[0].focus();
+    modal.el.addEventListener('keydown', (e) => {
+      this.trap(e, modal);
+    });
+  }
+
+  closeModal(modal) {
+    modal.el.classList.remove('modal--active');
```

Comment:
shouldn't we set the `modal--active` class in some variable? You use it a couple of time, maybe will be more readable?